File: resolve.py
import os
import sys

def resolve_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        lines = f.readlines()
    
    out_lines = []
    in_conflict = False
    in_head = False
    
    for line in lines:
        if line.startswith('<<<<<<< HEAD'):
            in_conflict = True
            in_head = True
            continue
        elif line.startswith('======='):
            in_head = False
            continue
        elif line.startswith('>>>>>>>'):
            in_conflict = False
            continue
            
        if not in_conflict:
            out_lines.append(line)
        elif in_head:
            out_lines.append(line)

    with open(filepath, 'w', encoding='utf-8') as f:
        f.writelines(out_lines)

files = [
    'frontend/src/components/Navbar.tsx',
    'frontend/src/pages/AboutPage.tsx',
    'frontend/src/pages/LandingPage.tsx',
    'frontend/src/pages/WalletPage.tsx'
]

for file in files:
    resolve_file(file)
    print(f"Resolved {file}")

File: resolve.js
const fs = require('fs');

const files = [
    'frontend/src/components/Navbar.tsx',
    'frontend/src/pages/AboutPage.tsx',
    'frontend/src/pages/LandingPage.tsx',
    'frontend/src/pages/WalletPage.tsx'
];

files.forEach(filepath => {
    const lines = fs.readFileSync(filepath, 'utf8').split('\n');
    const out_lines = [];
    let in_conflict = false;
    let in_head = false;

    for (let i = 0; i < lines.length; i++) {
        const line = lines[i];
        if (line.startsWith('<<<<<<< HEAD')) {
            in_conflict = true;
            in_head = true;
            continue;
        } else if (line.startsWith('=======')) {
            in_head = false;
            continue;
        } else if (line.startsWith('>>>>>>>')) {
            in_conflict = false;
            continue;
        }

        if (!in_conflict) {
            out_lines.push(line);
        } else if (in_head) {
            out_lines.push(line);
        }
    }

    fs.writeFileSync(filepath, out_lines.join('\n'), 'utf8');
    console.log(`Resolved ${filepath}`);
});
